DietPi-Software | Xserver uninstall fix + minor improvements

MichaIng opened the pull request:
**Status**: Ready for review and testing
- [x] Patch fix on v6.12 update
- [x] Add changelog entry

**Reference**: https://github.com/Fourdee/DietPi/issues/1921

**Commit list/description**:
+ DietPi-Software | Desktops: "iceweasel" is transitional package for "firefox-esr" on all repos. Switch to latter to avoid confusion, doubled menu/desktop entries etc: https://packages.debian.org/stretch/iceweasel
+ DietPi-Software | Xserver: Clean up manual installed packages based on dependencies, to avoid uninstall errors: https://github.com/Fourdee/DietPi/issues/1921
+ DietPi-Software | Reduce e.g. doubled install output a bid
+ DietPi-Patchfile | Fix Xserver uninstall issus by not purging dependencies, but leaving them for autoremove: #1921 
+ DietPi-Patchfile | Added G_RUN_CMD to software userdata `mv`, to prevent removing them in case of failure
+ CHANGELOG | Xserver: Resolved rarely occuring uninstall issus by not purging dependencies, but leaving them for autoremove
+ DietPi-Patchfile | Reset CurlFTPFS install state, which was removed with v6.10

MichaIng committed:
v6.12

+ DietPi-Software | Desktops: "iceweasel" is transitional package for "firefox-esr" on all repos. Switch to latter to avoid confusion, doubled menu/desktop entries etc: https://packages.debian.org/stretch/iceweasel
+ DietPi-Software | Xserver: Clean up manual installed packages based on dependencies, to avoid uninstall errors: https://github.com/Fourdee/DietPi/issues/1921
+ DietPi-Software | Reduce e.g. doubled install output a bid

MichaIng committed:
v6.12

+ DietPi-Patchfile | Fix Xserver uninstall issus by not purging dependencies, but leaving them for autoremove: https://github.com/Fourdee/DietPi/issues/1921

MichaIng committed:
v6.12

+ CHANGELOG | Xserver: Resolved rarely occuring uninstall issus by not purging dependencies, but leaving them for autoremove

MichaIng commented:
@Fourdee 
Other topic on dietpi-software:
- With v6.12 curlftpfs has been removed: https://github.com/Fourdee/DietPi/commit/814823899667e896291ed240281cfd39971463d9#diff-d92a6ee04e02fd2a2dc23d5bec3e6a98
- But install state was not reset: https://github.com/Fourdee/DietPi/blob/master/dietpi/patch_file#L763-L771
- This leads theoretically to 'NULL' entries within survey report and uninstall menu.
- Shall we catch up with this on v6.12 patch?

MichaIng committed:
Merge branch 'testing' into xserver_uninstall_fix

Fourdee commented:
@MichaIng 

> CURL, Shall we catch up with this on v6.12 patch?

Yep good spot, please patch for removal from installed file 👍

Fourdee reviewed (approved):
Good work, looks good 👍 

Merge when ready.

MichaIng committed:
v6.12

+ DietPi-Patchfile | Reset CurlFTPFS install state, which was removed with v6.10

MichaIng commented:
- [x] https://github.com/Fourdee/DietPi/pull/1930/commits/3e2c5d7d2bcf34dab77a319c03f2be6e0c3958ee
